api/task_status.go

```diff
@@ -42,26 +45,33 @@ func newTaskStatusHandler(store *event.Store, drivers *driver.Drivers, version s
 // ServeHTTP serves the task status endpoint which returns a map of taskname to
 // task status
 func (h *taskStatusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	log.Printf("[TRACE] (api.taskstatus) requesting task status '%s'", r.URL.Path)
+	logger := logging.FromContext(r.Context()).Named(taskStatusSubsystemName)
+	logger.Trace("request task status", "url_path", r.URL.Path)
 
 	taskName, err := getTaskName(r.URL.Path, taskStatusPath, h.version)
 	if err != nil {
-		log.Printf("[TRACE] (api.taskstatus) bad request: %s", err)
-		jsonErrorResponse(w, http.StatusBadRequest, err)
+		logger.Trace("bad request", "error", err)
+		if err = jsonErrorResponse(w, http.StatusBadRequest, err); err != nil {
+			logger.Error("error, could not generate json error response", "error", err)
+		}
 		return
 	}
 
 	filter, err := statusFilter(r)
 	if err != nil {
-		log.Printf("[TRACE] (api.taskstatus) bad request: %s", err)
-		jsonErrorResponse(w, http.StatusBadRequest, err)
+		logger.Trace("bad request", "error", err)
+		if err = jsonErrorResponse(w, http.StatusBadRequest, err); err != nil {
+			logger.Error("error, could not generate json error response", "error", err)
+		}
 		return
 	}
 
 	include, err := include(r)
 	if err != nil {
-		log.Printf("[TRACE] (api.taskstatus) bad request: %s", err)
-		jsonErrorResponse(w, http.StatusBadRequest, err)
+		logger.Trace("bad request", "error", err)
+		if err = jsonErrorResponse(w, http.StatusBadRequest, err); err != nil {
```

Comment:
There are 3 lines per the # of occurrences of this error handling. What do you think about passing the logger to the `jsonErrorResponse` helper function and do the err checking there?

Reply:
I agree, will do this